ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -56,13 +57,13 @@ constructor(
   private val surveyStore = localDataStore.surveyStore
 
   /**
-   * Emits the currently active survey on subscribe and on change. Emits `empty()]`when no survey is
-   * active.
+   * Emits the currently active survey on subscribe and on change. Emits `empty()`when no survey is
+   * active or local db isn't up-to-date.
    */
   val activeSurvey: @Cold Flowable<Optional<Survey>> =
-    localValueStore.activeSurveyIdFlowable.distinctUntilChanged().switchMapSingle {
-      if (it.isEmpty()) Single.just(Optional.empty())
-      else getOfflineSurvey(it).map { s -> Optional.of(s) }
+    localValueStore.activeSurveyIdFlowable.distinctUntilChanged().switchMapMaybe {
+      if (it.isEmpty()) Maybe.just(Optional.empty())
+      else maybeGetOfflineSurvey(it).map { s -> Optional.of(s) }
```

Comment:
I don't think we need to differentiate null/empty in the method names `maybeGetOfflineSurvey`. Once migrated to coroutines this will just be a suspend fun that returns `Survey?`. Or is there precedent for this?